openlibrary/catalog/add_book/__init__.py

```diff
@@ -600,8 +609,13 @@ def load(rec):
         e['ocaid'] = rec['ocaid']
         need_edition_save = True
 
-    # add values to edition lists
-    for f in 'source_records', 'local_id', 'ia_box_id', 'ia_loaded_id':
+
+    edition_fields = ['local_id', 'ia_box_id', 'ia_loaded_id']
```

Comment:
the intent of these modifications is to avoid clobbering `source_records` when a book is being updated, unless `source_records` is missing